src/components/Comment/Comment/Comment.jsx

```diff
@@ -1,67 +1,121 @@
+    useCommentContext();
+  const { deleteComment } = useComment();
+  const [isOptionModalOpen, setIsOptionModalOpen] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
 
-  const handleLikedClick = () => {
-    console.log('API로 liked 데이터 수정');
+  const onCommentOptionClick = ({ id, parentId, content }) => {
+    setCommentId(id);
+    setContent(content);
+    setIsOptionModalOpen(true);
   };
 
-  const handleCommentClick = () => {
-    if (onCommentClick) {
-      onCommentClick();
-    }
+  const onCloseClick = () => {
+    setCommentId(undefined);
+    setContent('');
   };
+  // const handleLikedClick = () => {
+  //   console.log('API로 liked 데이터 수정');
+  // };
 
   return (
-    <div className={styles.comment}>
-      <div className={styles.commentTop}>
-        <div className={styles.commentTopLeft}>
-          <div className={styles.cloud}>
-            <Icon id='cloud' width='19' height='13' />
+    <>
+      <div className={styles.comment} onClick={() => setCommentId(undefined)}>
+        <div className={styles.commentTop}>
+          <div className={styles.commentTopLeft}>
+            <div className={styles.cloud}>
+              <Icon id='cloud' width='19' height='13' />
+            </div>
+            <p>{data.userDisplay}</p>
+            <p className={styles.dot}>·</p>
+            <p>
+              {timeAgo(data.createdAt)} {data.isUpdated ? ' (수정됨)' : null}
+            </p>
           </div>
-          <p>{data.userDisplay}</p>
-          <p className={styles.dot}>·</p>
-          <p>
-            {timeAgo(data.createdAt)} {data.isUpdated ? ' (수정됨)' : null}
+          <p
+            className={styles.dot3}
+            onClick={(e) => {
+              e.stopPropagation();
+              onCommentOptionClick(data);
+            }}
+          >
+            {data.isWriter && (
+              <Icon id='ellipsis-vertical' width='3' height='11' />
+            )}
           </p>
         </div>
-        <p
-          className={styles.dot3}
-          onClick={() => onCommentOptionClick(data.id, data.content)}
-        >
-          <Icon id='ellipsis-vertical' width='3' height='11' />
-        </p>
-      </div>
-      <div className={styles.commentCenter}>{data.content}</div>
-      <div className={styles.commentBottom}>
-        <div className={styles.commentCount} onClick={handleCommentClick}>
-          <Icon id='comment' width='15' height='13' fill='#D9D9D9' />
-          <p>{data.children.length}</p>
-        </div>
-        <div className={styles.likedCount}>
-          <Icon id='like' width='13' height='12' fill='#D9D9D9' />
-          <p>{data.likeCount}</p>
+        <div className={styles.commentCenter}>{data.content}</div>
+        <div className={styles.commentBottom}>
+          <div
+            className={styles.commentCount}
+            onClick={(e) => {
+              e.stopPropagation();
+              setCommentId(data.id);
+              inputFocus();
+            }}
```

Comment:
추후 유지보수를 용이하게 하기 위해 onClick 의 콜백 함수 부분을 핸들러 함수로 따로 정의하는건 어떨까요?

```suggestion
            onClick={handleCommentCountButtonClick}
```